torchci/lib/bot/rerunGithubInfraErrorWorkflow.ts

```diff
@@ -4,10 +4,16 @@ const tagPrefix = "rerunGithubInfraFailure/";
 
 function rerunGithubInfraErrorWorkflow(app: Probot): void {
   let reruns: Map<string, number> = new Map();
+  let day = 0;
 
   // This bot is used to rerun failed workflows on pytorch/pytorch that look
   // like https://github.com/pytorch/pytorch/actions/runs/8454565307
   app.on("workflow_run.requested", async (ctx) => {
+    if (day != new Date().getDate()) {
+      // Reset reruns every day
```

Comment:
I see we currently allow up to 10 reruns. Do we have any stats on how often we actually hit that 10 rerun limit?

Really, I'm wondering how long it takes for a job to reach 10 reruns. Can jobs get into a state where they take longer than 24 hours to reach 10 reruns, where we'd clear the cache before they can hit their max, making it so that we'd rerun them forever?

Reply:
Instead of a daily cache purge, have you considered removing elements once the run successfully completes (like you do in https://github.com/pytorch/test-infra/pull/5059)?

That could be paired with a weekly cache purge on Mondays, reducing the risk of 10 reruns taking longer than 24 hours

Reply:
> I see we currently allow up to 10 reruns. Do we have any stats on how often we actually hit that 10 rerun limit?
I count about 25 so far
It is very unlikely for a hash to get 10+ more jobs after a 24 hour window, the way I can think of would be if someone triggered another workflow, which should be rerun regardless since its a new workflow